test-suite/src/lib.rs

```diff
@@ -111,3 +116,77 @@ pub fn create_temp_dir() -> String {
     std::mem::forget(td);
     path
 }
+
+/// Provides an available port (ranged from 37000 to 37999) for the test.
+pub async fn dispense_port() -> u16 {
+    use once_cell::sync::OnceCell;
+    static PORTS: OnceCell<RwLock<Vec<u16>>> = OnceCell::new();
+    PORTS
+        .get_or_init(|| RwLock::new((37000..38000).into_iter().collect::<Vec<_>>()))
+        .write()
+        .await
+        .pop()
+        .expect("wtf did we have tests more than 1000?")
+}
+
+pub async fn create_test_dms(
+    network_config: NetworkConfig,
+    dms_key: String,
+    peers: SharedKnownPeers,
+) -> Dms {
+    let path = create_temp_dir();
+
+    StorageImpl::create(&path).await.unwrap();
+    let storage = StorageImpl::open(&path).await.unwrap();
+    Dms::create(storage, 0, dms_key).await.unwrap();
+    let storage = StorageImpl::open(&path).await.unwrap();
```

Comment:
Suggest changing the signature of `Dms::create(mut storage: S, ...)` into `create(storage: &mut S, ...)` to avoid this pattern.